rest_framework_json_api/renderers.py

```diff
@@ -273,6 +273,30 @@ def extract_relationships(cls, fields, resource, resource_instance):
         return utils.format_keys(data)
 
     @classmethod
+    def extract_relation_instance(cls, field_name, field, resource_instance, serializer):
```

Comment:
Could you please add a docstring which provides a conceptual overview of what's going on here? I'm finding the four nested levels to be really intense. If you could use PEP 257 style, that would be great.

```python
"""Extract ...

<relevant details that might help understand what's going on here>
"""
```

This many levels of exceptions makes me wonder if there might performance challenges here (because of the cost of going through the exception stack over and over). I don't think it's fair to ask you for a performance test so I'm mostly wondering "out loud" if this is too much exception handling as a way to do control flow. It does feel like a bit of a code smell. Maybe that's the cost of working with a spec that's so dynamic ¯\\_(ツ)_/¯